src/main/java/org/myrobotlab/net/SslUtil.java

```diff
@@ -0,0 +1,90 @@
+package org.myrobotlab.net;
+
+import java.io.BufferedInputStream;
+import java.io.FileInputStream;
+import java.io.FileReader;
+import java.security.KeyPair;
+import java.security.KeyStore;
+import java.security.Security;
+import java.security.cert.CertificateFactory;
+import java.security.cert.X509Certificate;
+
+import javax.net.ssl.KeyManagerFactory;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.SSLSocketFactory;
+import javax.net.ssl.TrustManagerFactory;
+
+import org.bouncycastle.jce.provider.BouncyCastleProvider;
+import org.bouncycastle.openssl.PEMDecryptorProvider;
+import org.bouncycastle.openssl.PEMEncryptedKeyPair;
+import org.bouncycastle.openssl.PEMKeyPair;
+import org.bouncycastle.openssl.PEMParser;
+import org.bouncycastle.openssl.jcajce.JcaPEMKeyConverter;
+import org.bouncycastle.openssl.jcajce.JcePEMDecryptorProviderBuilder;
+
+public class SslUtil {
```

Comment:
please add a javadoc description to this class.

Reply:
one thing to keep in mind is that bouncycastle libraries have export restrictions which means that this code might not legally be able to be used in china or russia.  (and other countries.)  I'm not sure if that's changed in more recent times.

Reply:
Ahahahah .. better tell Amazon to shut down their Bejing IoT Core ...
https://docs.amazonaws.cn/en_us/aws/latest/userguide/iot.html#feature-regions

Fixed